### PR TITLE
Support all keyboard actions. (#11344)

### PR DESCRIPTION
**Pull request by matthew-carroll**
Support all keyboard actions for iOS and Android. (#11344)

There is a corresponding framework PR:
https://github.com/flutter/flutter/pull/18855

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 1]
```diff
@@ -277,14 +315,52 @@ - (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
 }
 
 - (BOOL)shouldChangeTextInRange:(UITextRange*)range replacementText:(NSString*)text {
-  if (self.returnKeyType == UIReturnKeyDone && [text isEqualToString:@"\n"]) {
-    [self resignFirstResponder];
-    [self removeFromSuperview];
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Are these not still needed in the done case?

Comment (matthew-carroll) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: I'm not 100% sure.  My experimentation with an actual app didn't show any problem without them, and my assumption is that when the Flutter side says "unfocus" that those same methods are being invoked in some other execution path.  But I'm not positive.

Comment (matthew-carroll) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: FYI - @cbracken and I looked at Japanese and "done" style actions, as well as "next" style actions appear to work as desired with these lines removed.

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 2]
```diff
@@ -30,9 +31,46 @@ static UIKeyboardType ToUIKeyboardType(NSDictionary* type) {
 }
 
 static UIReturnKeyType ToUIReturnKeyType(NSString* inputType) {
-  if ([inputType isEqualToString:@"TextInputType.multiline"])
+  // Where did the term "unspecified" come from? iOS has a "default" and Android
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Thanks for the extra context doc!!

**Comment by xster** [comment]
LGTM

**Pull request merged**